File: app/src/main/java/com/projectfox/foxoff/brain/FoxBrainAnalyzer.kt
package com.projectfox.foxoff.brain

/**
 * Contract for engines capable of calculating sleep scores from events.
 */
interface FoxBrainAnalyzer {
    /**
     * Analyses an event relative to the current brain state and returns a new score.
     */
    fun analyze(event: FoxBrainEvent, currentState: FoxBrainState): FoxBrainScore
}

File: ios/FoxCoreKit/Sources/FoxCoreKit/FoxBrainAnalyzer.swift
/// Portage direct de `com.projectfox.foxoff.brain.FoxBrainAnalyzer` (Kotlin).
/// Contrat pour les moteurs capables de calculer un score de sommeil à
/// partir d'événements.
public protocol FoxBrainAnalyzer {
    /// Analyse un événement par rapport à l'état courant et renvoie un
    /// nouveau score.
    func analyze(_ event: FoxBrainEvent, currentState: FoxBrainState) -> FoxBrainScore
}
